internal/controller/drplacementcontrol_controller.go

```diff
@@ -566,7 +566,7 @@ func GetDRClusters(ctx context.Context, client client.Client, drPolicy *rmn.DRPo
 func (r DRPlacementControlReconciler) updateObjectMetadata(ctx context.Context,
 	drpc *rmn.DRPlacementControl, placementObj client.Object, log logr.Logger,
 ) error {
-	update := false
+	var update bool
```

Comment:
What is the issue with the previous code? Both are valid Go code and should compile to the same thing. If not this should be a Go compiler bug.

Maybe we need to report a golangci-lint bug for this, the linter should not requiring changing such code.

Reply:
It is a wasted assignment as it is overwritten right after. The linter is for a code practice not a code error afaict. I do not see an issue with the same.

Reply:
There is already an issue this:
https://github.com/sanposhiho/wastedassign/issues/39

Reply:
I think I did visit that issue, once addressed it may help. For now it is better to have the linter than not, so I suggest we do not wait for a resolution of that issue (which seems to be 2+ years old).